### PR TITLE
Fix branching

### PR DESCRIPTION
**Pull request by ndmckinley**
The 'get' actually resolves the version _before_ the branch is chosen,
which means it can't know what the branch is, which means there's no
way for it work with the 'try / on_success' model.

Instead, I wrote some really simple python which gets existing
depends: ... comments.

I've tested it on PR #43 and it works.

-----------------------------------------------------------------
# [all]
CI Changes only.

### CODE REVIEW

File: .ci/magic-modules/branch.yml [hunk 1]
```diff
@@ -12,8 +12,8 @@ image_resource:
         # bash available, we can switch to that, but it's probably better to
         # minimize the number of containers used than to minimize the size of
         # the container in this particular (very short) task.
-        repository: nmckinley/go-ruby
-        tag: '1.9-2.5'
+        repository: nmckinley/python
```
Comment (danawillow) [inline .ci/magic-modules/branch.yml]: can you update the comments above this?

File: .ci/magic-modules/get_downstream_prs.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python
+import functools
+import os
+import re
+import sys
+from github import Github
+
+if __name__ == '__main__':
+  g = Github(os.environ.get('GH_TOKEN'))
+  all_prs = functools.reduce(lambda x,y: x+re.findall(r'\ndepends: (https://github.com/.*)', y.body),
```
Comment (danawillow) [inline .ci/magic-modules/get_downstream_prs.py]: I'm not super familiar with python style, but is this typical? It feels like a lot is being crammed into these two lines.  If it is, I wouldn't mind seeing a quick comment that explains what's going on.

File: .ci/magic-modules/get_downstream_prs.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+      g.get_repo('GoogleCloudPlatform/magic-modules').get_pull(int(sys.argv[1])).get_issue_comments(), [])
```
Comment (danawillow) [inline .ci/magic-modules/get_downstream_prs.py]: I don't expect pagination to actually be an issue, but do you think it's worth it to just handle it now so it doesn't bite us later?

File: .ci/magic-modules/branch.yml (hunk 1 above)
Comment (ndmckinley) [reply .ci/magic-modules/branch.yml]: Thanks, good catch.

**Pull request merged**